### PR TITLE
Attempt to fix redirected PUT requests not working on Andorid

### LINKED ISSUE #7058: Redirected PUT requests fails on Online Multiplayer
**Platform**
Android 5.1.1 (Tester @touhidurrr)
Android 8.0.0 (Tester @letstalkaboutdune)
Android 12 (Tester @alexban011)

**Version**
4.1.5 - 4.1.6

**Describe the bug**
If a multiplayer server url redirects to another url then put requests shows failure and create new game doesn’t work.
For tests: current state of https://uncivserver.xyz/

### PR DESCRIPTION
**Pull request by touhidurrr**
Might resolve #7058

### CODE REVIEW

**Comment by touhidurrr** [comment]
Lol. I opened this by mistake. I was thinking of using the pen button as a shortcut to edit the file and open a new branch.
Anyways @Azzurite I just realised that it won't work cause no way to read headers from SimpleHTTP.
Related to: https://github.com/yairm210/Unciv/issues/7058

**Comment by touhidurrr** [comment]
> Maybe you know how to do it since you wrote it?

Oh mistake. It is @SomeTroglodyte

**Comment by touhidurrr** [comment]
This doesn't break things on PC according to my tests but not sure if it solves the problems on Android. @alexban011 can test This PR and without this PR on Android on server `https://uncivserver.xyz` ?

**Comment by alexban011** [comment]
Tested. The patch works.

**Comment by touhidurrr** [comment]
> Tested. The patch works.

You sure this works on Android?

**Comment by touhidurrr** [comment]
Good! Thanks for helping!! Then maybe I will keep this open.

**Comment by touhidurrr** [comment]
> Oh, after I click next turn it says waiting for .... (your civ)
Then when trying to join the game again it says file not found on the mp server

Lol, do you know how to fix this? I am not familiar with Kotlin codes. In Katlin sometimes people just doesn’t return anything in a function. Dunno how that works.

**Comment by alexban011** [comment]
what version is the server? does it include  #7095 aka 4.1.8?

**Comment by alexban011** [comment]
redirects do happen, I get file not found in at https://uncivserver.herokuapp.com/files/bca29e9f-6019-4f03-af0d-b9285639f860 even though my server is uncivserver.xyz

File: core/src/com/unciv/logic/multiplayer/storage/SimpleHttp.kt [hunk 1]
```diff
@@ -46,7 +46,8 @@ object SimpleHttp {
                     outputStream.flush()
                 }
 
-                val text = BufferedReader(InputStreamReader(inputStream)).readText()
+                val redirected = responseCode == 301 || responseCode == 302 || responseCode == 303
```
Comment (alexban011) [inline core/src/com/unciv/logic/multiplayer/storage/SimpleHttp.kt:49]: I get responseCode 404, so make sure you have the latest version for the server @touhidurrr

**Pull request closed without merging**